lib/WUI/nhttp/gcode_upload.cpp

```diff
@@ -113,6 +117,14 @@ GcodeUpload::UploadResult GcodeUpload::start(const RequestParser &parser, Upload
         return StatusPage(Status::LengthRequired, StatusPage::CloseHandling::ErrorClose, json_errors);
     }
 
+    const char *path = holds_alternative<PutParams>(uploadParams) ? get<PutParams>(uploadParams).filepath.data() : nullptr;
+    auto slot = Monitor::instance.allocate(Monitor::Type::Link, path, *parser.content_length);
```

Comment:
Just to confirm (I think it is correct), but does this content-length correspond to the correct values in case of the Post, where this is the size of the form, not the file? That is, the `progress` is being called with the undecoded data?

Reply:
I know it is the length of the whole form. I'm just making sure this size corresponds to the values that are being reported by `progress` below ‒ that is, that `progress` is being called _before_ decoding the form and will not terminate at 95% or so, because the latter included only the „real“ data.